controllers/analysis_step.go

```diff
@@ -143,20 +143,17 @@ func (step *AnalysisStep) cleanUpStaleResults(rawResults map[string]corev1alpha1
 }
 
 func (step *AnalysisStep) processRawResults(rawResults map[string]corev1alpha1.Result, instance *K8sGPTInstance) error {
+
+	m := instance.r.MetricsBuilder.GetGaugeVec("k8sgpt_number_of_results_by_type")
+	if m != nil {
+		m.Reset()
+	}
 	for _, result := range rawResults {
-		operation, err := resources.CreateOrUpdateResult(instance.ctx, instance.r.Client, result)
+		_, err := resources.CreateOrUpdateResult(instance.ctx, instance.r.Client, result)
```

Comment:
Is it still necessary for the function to return the `operation` type since we no longer use it?